executor.go

```diff
@@ -71,65 +69,26 @@ func (x *executor) Execute(req *Request, re ResponseEmitter, env Environment) (e
 			re, res = NewChanResponsePair(req)
 
 			go func() {
-				var closeErr error
-
 				defer close(errCh)
-
-				err := cmd.PostRun[typer.Type()](res, lower)
-				closeErr = lower.CloseWithError(err)
-				if closeErr == ErrClosingClosedEmitter {
-					// ignore double close errors
-					closeErr = nil
-				}
-				errCh <- closeErr
-
-				if closeErr != nil && err != nil {
-					log.Errorf("error closing connection: %s", closeErr)
-					log.Errorf("close caused by error: %s", err)
-				}
+				errCh <- lower.CloseWithError(cmd.PostRun[typer.Type()](res, lower))
 			}()
 		}
 	} else {
 		// not using this channel today
 		close(errCh)
 	}
 
-	defer func() {
```

Comment:
We really shouldn't be catching panics this way. This only catches panics where the panic value is an _error_ so it's kind of arbitrary anyways.

IIRC, this was added because we had a bunch of races around `Close` and `SetError` but those have since been fixed.